api/src/main/scala/za/co/absa/loginsvc/rest/config/jwt/AwsSecretsManagerKeyConfig.scala

```diff
@@ -36,18 +36,49 @@ case class AwsSecretsManagerKeyConfig(
   accessExpTime: FiniteDuration,
   refreshExpTime: FiniteDuration,
   pollTime: Option[FiniteDuration],
+  keyLayOverTime: Option[FiniteDuration],
   keyPhaseOutTime: Option[FiniteDuration]
 ) extends KeyConfig {
 
   private val logger = LoggerFactory.getLogger(classOf[AwsSecretsManagerKeyConfig])
 
   override def keyRotationTime : Option[FiniteDuration] = pollTime
-  override def keyPair(): (KeyPair, Option[KeyPair]) = {
+  override def keyPair(): (KeyPair, Option[KeyPair]) = fetchKeySetsFromCloud()
+
+  override def throwErrors(): Unit = this.validate().throwOnErrors()
+
+  override def validate(): ConfigValidationResult = {
+
+    val awsSecretsResults = Seq(
+      Option(secretName)
+        .map(_ => ConfigValidationSuccess)
+        .getOrElse(ConfigValidationError(ConfigValidationException("secretName is empty"))),
+
+      Option(region)
+        .map(_ => ConfigValidationSuccess)
+        .getOrElse(ConfigValidationError(ConfigValidationException("region is empty"))),
+
+      Option(privateKeyFieldName)
+        .map(_ => ConfigValidationSuccess)
+        .getOrElse(ConfigValidationError(ConfigValidationException("privateKeyFieldName is empty"))),
+
+      Option(publicKeyFieldName)
+        .map(_ => ConfigValidationSuccess)
+        .getOrElse(ConfigValidationError(ConfigValidationException("publicKeyFieldName is empty"))),
+    )
+
+    val awsSecretsResultsMerge = awsSecretsResults.foldLeft[ConfigValidationResult](ConfigValidationSuccess)(ConfigValidationResult.merge)
+
+    super.validate().merge(awsSecretsResultsMerge)
+  }
+
+  def fetchKeySetsFromCloud(secretsUtils: SecretUtil = AwsSecretsUtils): (KeyPair, Option[KeyPair]) = {
```

Comment:
```suggestion
/**
   * Some description here
   * @param secretsUtils
   * @return tuple of [AWSCURRENT keyPair, optionally AWSPREVIOUS]
   */
  private[jwt] def fetchKeySetsFromCloud(secretsUtils: SecretUtil = AwsSecretsUtils): (KeyPair, Option[KeyPair]) = {
```

Suggesting: limiting visibility (essentially private but available for tests), 
simple doc to signify which key is which (since they are of the same type)